apps/dashboard/app/views/batch_connect/session_contexts/_form.html.erb

```diff
@@ -6,6 +6,46 @@
   <%= f.submit t('dashboard.batch_connect_form_launch'), class: "btn btn-primary btn-block" %>
 <% end %>
 
+<%= javascript_tag do %>
+(function(){
+	'use strict'
+
+	/**
+	 * Toggle the visibilty of a form group
+	 */
+	function toggle_visibilty_of_elements() {
+	  /* array of controlled elements */
+      let controlled_form_groups = Array.from(document.querySelectorAll('.form-group[data-ood-show-for]'));
+
+      /* array of active scopes' values */
+      let enabled_scopes_values = Array.from(document.querySelectorAll('option[data-ood-show-scope]:checked, input[data-ood-show-scope]:checked').values()).map(x => x.dataset.oodShowScope);
+
+
+      controlled_form_groups.forEach(el => update_form_group_visibility(el, enabled_scopes_values.includes(el.dataset.oodShowFor)));
+	}
+
+	/*
+    * Updates the visibility of controlled a form group
+    */
+    function update_form_group_visibility(e1, visible) {
+      let input_element = $(e1).find(":input");
+      visible ? $( e1 ).show() && input_element.prop( "disabled", false ) : $( e1 ).hide() && input_element.prop( "disabled", true );
```

Comment:
you can use $( el ).toggle(visible) to get rid of the control flow i.e.

```
$( el ).toggle(visible);
input_element.prop( "disabled", ! visible );
```

Reply:
And since its only 2 lines you could probably inline with the other:

```
controlled_form_groups.forEach(el => {
  let input_element = $(e1).find(":input");
  let visible = enabled_scopes_values.includes(el.dataset.oodShowFor);
  
  $( el ).toggle(visible);
  input_element.prop( "disabled", ! visible );
}
```